package protoFile.quest.getQuestListOnNpc;
message getQuestListOnNpcRequest{
     required int32 id = 1;
     required int32 npcId = 2;
}

message getQuestListOnNpcResponse{
     required bool result = 1;
     required string message = 2;
     optional ResponseData data = 3; 
}

message ResponseData{
     required int32 npcId = 1;
     repeated Quest canReceivedquestList = 2;    //可接受的任务列表
     repeated Quest progressingquestList = 3;    //进行中的任务列表   
     repeated Quest cancommitquestList = 4;      //可提交的任务列表
     repeated Quest failedquestList = 5;	 //失败了的任务列表
}

message Quest{
     optional int32 taskId = 1;			//任务的ID
     optional string taskname = 2;		//任务的名称
}
